cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -704,6 +704,22 @@ def _validate_dataframe(self, df_name: str):
                 column_def = self._get_column_def(df_name, column_name)
                 column = getattr(df, column_name)
 
+                # Validate that {column}_colors exists)
```

Comment:
mostly not sure if this is the right entry point

Reply:
`{column}_colors` is in the `AnnData.uns` attribute (which is a dictionary of unstructured data, i.e. data that isn't indexed by cells or genes). I think you're currently assuming that `{column}_colors` is in a dataframe.

Reply:
Take a look [here](https://github.com/chanzuckerberg/single-cell-curation/blob/09e545bb6e748088acc45687bc3ad0073e4a0496/cellxgene_schema_cli/cellxgene_schema/schema_definitions/3_0_0.yaml#L17) for the schema components. You can see that `uns` is a `dict`, which means you need `_validate_dict`.

Reply:
thanks! this is super helpful